src/snyk/common/views/issueTreeProvider.ts

```diff
@@ -119,6 +119,23 @@ export abstract class ProductIssueTreeProvider<T> extends AnalysisTreeNodeProvid
     return false; // optionally overridden by products
   }
 
+  filterVisibleIssues(issues: Issue<T>[]): Issue<T>[] {
+    return issues.filter(issue => this.isVisibleIssue(issue, this.configuration.issueViewOptions))
+  }
+
+  protected isVisibleIssue(_issue: Issue<T>, issueViewOptions: IssueViewOptions) {
+    if (issueViewOptions.ignoredIssues && issueViewOptions.openIssues){
+      return true
+    }
+    if (issueViewOptions.ignoredIssues) {
+       return _issue.isIgnored == true
+    }
+    if (issueViewOptions.openIssues){
+      return _issue.isIgnored != true
+    }
+    return false
+  }
```

Comment:
non-blocking: if it's okay with you, maybe we could name the conditions to understand a bit better when they get shown? Something like:

```suggestion
  filterVisibleIssues(issues: Issue<T>[]): Issue<T>[] {
    return issues.filter(issue => this.isVisibleIssue(issue, this.configuration.issueViewOptions));
  }

  protected isVisibleIssue(issue: Issue<T>, issueViewOptions: IssueViewOptions): boolean {
    const { ignoredIssues: ignoredIssuesOpt, openIssues: openIssuesOpt } = issueViewOptions;
    const isIssueIgnored = issue.isIgnored;

    // Show all issues
    if (ignoredIssuesOpt && openIssuesOpt) {
      return true;
    }

    // Show only ignored or only issues based on settings
    return (ignoredIssuesOpt && isIssueIgnored) || (openIssuesOpt && !isIssueIgnored);
  }
```

- I added this comment to the change I'm suggestion but is is correct: `// Show only ignored or only issues based on settings`? Is it what we mean?

If you think it's not worth it, I'd suggest to at least add the missing semicolons and change to strict equality? `===` instead og `==`. But in this case we could just return the value of `_issue.isIgnored` since it's already a boolean, as far as I understand.